modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java

```diff
@@ -32,6 +36,8 @@
 import org.apache.synapse.task.TaskManager;
 import org.apache.synapse.task.TaskManagerObserver;
 
+import sun.misc.Service;
```

Comment:
Avoid using low level private APIs such ash sun.misc.Service, which can lead to interoperability issues if the same code is run on a JDK introduced by a different vendor. The same API was moved into a public API in all new JDK versions, namely, ServiceLoader, which can effectively be used as an alternative.